src/framework/types.ts

```diff
@@ -186,6 +186,19 @@ export namespace OpenAPIV3 {
     externalDocs?: ExternalDocumentationObject;
   }
 
+  interface ComponentsV3_1 extends ComponentsObject {
+    pathItems?: { [path: string]: PathItemObject | ReferenceObject }
+  }
+
+  export interface DocumentV3_1 extends Omit<DocumentV3, 'paths' | 'info' | 'components'> {
```

Comment:
```suggestion
  export interface DocumentV3_1 extends Omit<DocumentV3, 'paths' | 'info' | 'components', 'openapi'> {
      openapi: `3.1.${p}`;
```
To make this also compile type safe:
could then also be applied above in 3.0 with `openapi: `3.0.${p}`;`